dotnet/src/Connectors/Connectors.AssemblyAI/AssemblyAIServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.SemanticKernel.AudioToText;
+using Microsoft.SemanticKernel.Connectors.AssemblyAI;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides extension methods for <see cref="IServiceCollection"/> and related classes to configure AssemblyAI connectors.
+/// </summary>
+public static class AssemblyAIServiceCollectionExtensions
+{
```

Comment:
These extensions are too complicated. I don't think using an options is needed. Leave only two methods, first for kernelBuilder and second for serviceCollection. Service extensions should also allow passing HttpClient by end developer. As example first two methods: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceCollectionExtensions.cs Only pass apiKey, endpoint and httpclient. Polling interval pass as execution settings like before.

Reply:
This is usually how I set up my libraries, following [this guidance for library authors](https://learn.microsoft.com/en-us/dotnet/core/extensions/options-library-authors).
I'm not sure why it would be different here, but I'll simplify it for now.

Reply:
We can consider add more extensions to DI but then we also should extend others connectors to keep things consistent. But for now I'd like to minimize less important code, we can back to that when the assemblyAI branch will be ready for merge to main.